core/src/main/java/com/linecorp/armeria/server/RoutingTrie.java

```diff
@@ -223,16 +223,24 @@ private Node<V> checkNode(Node<V> node, String path, int begin, boolean exact, I
                 break;
             case PARAMETER:
                 // Consume characters until the delimiter '/' as a path variable.
-                final int delim = path.indexOf('/', begin);
-                if (delim < 0) {
-                    // No more delimiter.
-                    return node;
+                final int delimSlash = path.indexOf('/', begin);
+                final int lastColon = path.lastIndexOf(':');
+
+                if (delimSlash < 0) {
+                    if (lastColon < 0) {
+                        // No more delimiter.
+                        return node;
+                    } else {
```

Comment:
`path.lastIndexOf(':')` does not handle `:` character in the path segment such as `/foo:bar/baz`.
I think `lastColon` should be searched from `begin`.
Additionally, `lastColon` can be checked only if `delimSlash < 0`
```suggestion
                if (delimSlash < 0) {
                    final int lastColon = path.lastIndexOf(':', begin);
                    // lastIndexOf do backward search
                    if (lastColon < begin) {                
                        // No more delimiter.
                        return node;
                    } else {
```
Should we also add some test cases using annotated services for that?